src/Extensions/SiteTreeExtension.php

```diff
@@ -268,52 +317,79 @@ public function MetaTags(&$tagString)
+                        'attributes' => [
+                            'property' => 'article:author_' . $key,
+                            'content' => $openGraphAuthor,
+                        ],
+                    ];
                 }
             }
 
             if ($openGraphPublicationTime) {
-                $socialMetaTags[] = HTML::createTag('meta', [
-                    'property'  =>  'article:published_time',
-                    'content'   =>  $openGraphPublicationTime
-                ]);
+                $tags['article:published_time'] = [
+                    'attributes' => [
+                        'property' => 'article:published_time',
+                        'content' => $openGraphPublicationTime,
+                    ],
+                ];
             }
 
             if ($openGraphModificationTime) {
-                $socialMetaTags[] = HTML::createTag('meta', [
-                    'property'  =>  'article:modification_time',
-                    'content'   =>  $openGraphModificationTime
-                ]);
+                $tags['article:modification_time'] = [
+                    'attributes' => [
+                        'property' => 'article:modification_time',
+                        'content' => $openGraphModificationTime,
+                    ],
+                ];
             }
 
             if ($openGraphSection) {
-                $socialMetaTags[] = HTML::createTag('meta', [
-                    'property'  =>  'article:section',
-                    'content'   =>  $openGraphSection
-                ]);
+                $tags['article:section'] = [
+                    'attributes' => [
+                        'property' => 'article:section',
+                        'content' => $openGraphSection,
+                    ],
+                ];
             }
 
             if ($openGraphTags && is_array($openGraphTags)) {
-                foreach ($openGraphTags as $openGraphTag) {
-                    $socialMetaTags[] = HTML::createTag('meta', [
-                        'property'  =>  'article:tag',
-                        'content'  =>  $openGraphTag
-                    ]);
+                foreach ($openGraphTags as $key => $openGraphTag) {
+                    $tags['article:tag_' . $key] = [
+                        'attributes' => [
+                            'property' => 'article:tag',
+                            'content' => $openGraphTag,
+                        ],
+                    ];
                 }
             }
         }
 
+
+        if ($this->owner->hasMethod('updateSocialMetaTags')) {
+            $socialMetaTags = $this->owner->updateSocialMetaTags($socialMetaTags);
+        }
+
+        // todo figure out
+        //$extraMeta = $this->owner->getSocialMetaValue('ExtraMeta');
+        //if ($extraMeta) {
+        //$tagString .= "\n" . $extraMeta;
+        //}
+
+    }
+
+    public function MetaTags(&$tagString)
+    {
         $schemaData = null;
```

Comment:
could we move the schema data to MetaComponents as well?

Reply:
I'll explore this further. I had that initial thought as well but opted to put in the PR and leave that as is. I'm not sure how the components to tags will work with this specifically at this point, however the ability do do the same key/val pairing would definitely be useful here.

Reply:
The individual parts of the `json+ld` schema can be edited in `SiteConfig::updateSchemaData()` or `Page::getSocialMetaSchemaData()` respectively. We should be able to move the adding of the schema data to `MetaComponents` using something like this:

```
$tags['title'] = [
    'tag' => 'script',
    'attributes' => [
        'type' => 'application/ld+json',
    ],
    'content' => json_encode($schemaData, $options)
];
```